### DISCUSSION TITLE
How to release an app on Google Play?

### DISCUSSION THREAD

**Discussion by wschume** [flet-dev/flet#3360] [Q&A] [answered]
### Question

I have finished my first app I wish to publish and created an account on Google Play Console in order to publish the app. After building the app in the Android App Bundle mode (aab) I tried to upload the app to Google Play Console. Here I get the error message:

You uploaded an APK or Android App Bundle that was signed in debug mode. You need to sign your APK or Android App Bundle in release mode.

![image](https://github.com/flet-dev/flet/assets/65328438/897a779b-37ad-4969-9c81-ccba270e9e8e)

What am I missing to publish the app on Google Play?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by CyberTuz** [comment, marked as answer]
I had the same issue, you need to sign the app using a keystore file, which you can generate either through Android Studio or using the "keytool" utility.
After generating the keystore file, I placed it in "../build/flutter/android/app" directory.
Then, I used this command to build and sign the app: 
`flet build aab --android-signing-key-store FILE.jks --android-signing-key-alias ALIAS --android-signing-key-store-password PASSWORD --android-signing-key-password PASSWORD`

This worked for me and the AAB file was accepted on the Google Play Console .

**Reply by kumar2215** [reply]
Hi, I had the same issue too. This fixed it but now I get this:

Remove conflicts from the manifest before uploading. The following content provider authorities are in use by other developers: com.flet.src.androidx-startup, com.flet.src.com.pairip.licensecheck.LicenseContentProvider.
You need to use a different package name because "com.flet.src" already exists in Google Play.

I have tried making modifications to the .toml file and rebuilt multiple times but still failed. If you faced a similar issue or know how to fix it, please do tell me.

**Reply by CyberTuz** [reply]
Hi, I haven't had this error. 
But I guess this is caused because you didn't enter your package information and so by default it's using the Flet one, that is already present on the play store, but it's just a guess.

I have the following parameters on my pyproject.toml:

[tool.flet]
org = "com.company"
product = "My app"
company = "My company"

[tool.flet.android]
package_name = "com.company.myapp"
build_mode = "release"

On Android the package starts with the domain first and then the name, like in the example. 
I hope this can help you.

**Reply by kumar2215** [reply]
I tried this and used the [tool.flet.android] header in my .toml file but after rebuilding, the "com.flet.src" still appears as the applicationId in the build.gradle file inside "../build/flutter/android/app" directory causing the same error. Is there something I am not doing right?

**Reply by CyberTuz** [reply]
There might be some cached information causing issues, on my build.gradle are present the info from my pyproject.toml. 
You could try adding '--clear-cache' on the build command, or if that doesn't work, you could either wipe out the build folder or just copy your code over to a fresh Flet project, and see if the problem persists.
